src/PublisherMethods.js

```diff
@@ -103,5 +105,37 @@ module.exports = {
         _.nextTick(function() {
             me.trigger.apply(me, args);
         });
-    }
+    },
+
+    /**
+     * Returns a Promise for the triggered action
+     */
+    triggerPromise: function(){
+        var me = this;
+        var args = arguments;
+
+        var canHandlePromise =
+            this.children.indexOf('completed') >= 0 &&
+            this.children.indexOf('failed') >= 0;
+
+        if (!canHandlePromise){
+            throw new Error('Publisher must have "completed" and "failed" child publishers');
+        }
+
+        var promise = new Promise(function(resolve, reject) {
```

Comment:
If this line is using a factory method in `utils` for creating promises of the user's (Bluebird, Q or native), I will be more happy with this solution since I don't want to dictate (at the moment) what promise library users should be using. I.e. this line should be something like

```javascript
var promise = utils.createPromise(function(resolve, reject) { 
  //...
```

That way anyone can patch `createPromise` with whatever or use native promise shim of their choice. This also opens for the possibility for reflux to be extended with more promise based helper methods.